packages/manager/modules/vps/src/upgrade/vps-upgrade.controller.js

```diff
@@ -87,7 +87,7 @@ export default class VpsUpgradeCtrl {
       modelVersion,
       versionInfos.year < 2018 ? '2018v3' : '2018v4',
     );
-    const mappedType = get(OFFER_AGORA_MAPPING, modelType, modelType);
+    const mappedType = get(OFFER_AGORA_MAPPING, modelType, modelType == 'ssd' && (modelVersion == "2017v1" || modelVersion == "2018v2") ? 'ssd-discovery' : modelType);
```

Comment:
Retrieving the type is getting harder.

Why not considering having a method in a service for this?